docs/COMPONENT_GUIDELINES.md

```diff
@@ -29,3 +29,80 @@ So solutions like `@media-queries` might not work, and `window` shouldn't be dir
 
 * `wix-ui-tpa` are rendered for Wix's users-of-users, and so they must comply with web accessibility
  principles.
+ 
+ ## Typography
+ When creating a new component that renders text, we should in most cases use the `<Text/>` component with the correct props and style.  
+ In some cases we might need to use a typography that isn't defined in the `<Text/>` component.  
+ In this case, we rely on the style processor's syntax, to ensure we get the correct font from the viewer.
+ 
+ 
+ ### How it works
+ We use the [style processor's syntax](https://github.com/wix/wix-style-processor) to get our components wired correctly:  
+ The platform (viewer/editor) references the different fonts by a theme name, such as `Body-M` for paragraph text, and `Heading-L` for titles.  
+ To use these fonts we write for instance:
+ ```css
+:import {
+  -st-from: "src/common/formatters.st.js";
+  -st-named: font;
+}
+
+.myClass {
+  font: font({theme: 'Body-M', size: '16px', weight: 'bold'});
```

Comment:
don't you need to extend the `<Text/>`  component?

Reply:
I wish, but it doesn't work well at the moment, and it's not optimized for better performance.